lib/ngx/re.lua

```diff
@@ -168,6 +152,24 @@ function _M.split(subj, regex, opts, ctx, max, res)
                 break
             end
 
+            if from == to then
+                local old_pos = ctx.pos
+
+                local from2, _, _, err2 = re_split_helper(subj, compiled,
+                                              compile_once, flags, ctx)
```

Comment:
@thibaultcha It appears clearer to me now. Now I'd just like to be this match's result not wasted.